### PR TITLE
Abstract detector and Resolution fixes

### PR DESCRIPTION
**Pull request by marcus-oscarsson**
Hi,

Two minor fixes to AbstractDetector and Resolution.

All the best,
Marcus

### CODE REVIEW

File: HardwareObjects/abstract/AbstractDetector.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def init(self):
         """Initialise some common paramerters"""
 
         try:
-            self._metadata = self["beam"]
+            self._metadata = dict(self["beam"].getProperties())
```
Comment (beteva) [inline HardwareObjects/abstract/AbstractDetector.py:73]: This for some reason did not work the last time when I've tested (before the confinement). That's why I've used self["beam"] instead.

Comment (marcus-oscarsson) [reply HardwareObjects/abstract/AbstractDetector.py:73]: I see, it seems to work fine now :)

Comment (rhfogh) [reply HardwareObjects/abstract/AbstractDetector.py:73]: Could it be that getProperties osmetimes returns None, if the properties are not set?

Comment (marcus-oscarsson) [reply HardwareObjects/abstract/AbstractDetector.py:73]: In that case self["beam"] would give AttributeError

Comment (marcus-oscarsson) [reply HardwareObjects/abstract/AbstractDetector.py:73]: Actually it gives KeyError, the current code is fine :)

**Comment by marcus-oscarsson** [comment]
So it seems that I can merge it then :)

**Comment by marcus-oscarsson** [comment]
Let me know if I commit a terrible crime as I merge my own PR, but it seems like its approved ? :)

**Pull request merged**